proto/cosmos/tx/v1beta1/tx.proto

```diff
@@ -142,6 +142,11 @@ message AuthInfo {
   //
   // Since: cosmos-sdk 0.46
   Tip tip = 3;
+
+  // extension_options are arbitrary options that can be added by chains
+  // when the default options are not sufficient. If any of these are present
+  // and can't be handled, the transaction will be rejected
+  repeated google.protobuf.Any extension_options = 1023;
```

Comment:
> So we go back to re-use the TxBody.extension_options.

I actually agree with removing this field, and finally using TxBody's existing one. I think having `extension_options` on both AuthInfo and TxBody is confusing. Even though fees are not semantically 100% part of TxBody, using that existing field is still preferable, to have one single place to add extensions to a Tx.

Reply:
done, removed.
And changed `RejectExtensionOptionsMiddleware` to `NewExtensionOptionsMiddleware` which could allow some extension options now.